webserver/webserver/src/test/java/io/helidon/webserver/MtlsTest.java

```diff
@@ -0,0 +1,151 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.helidon.webserver;
+
+import io.helidon.common.configurable.Resource;
+import io.helidon.common.pki.KeyConfig;
+import io.helidon.config.Config;
+import io.helidon.config.MapConfigSource;
+import io.helidon.webclient.WebClient;
+import io.helidon.webclient.WebClientTls;
+import io.netty.handler.codec.DecoderException;
+import org.junit.jupiter.api.AfterAll;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import java.security.cert.X509Certificate;
+import java.util.HashMap;
+import java.util.Optional;
+import java.util.concurrent.ExecutionException;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.regex.Pattern;
+import javax.net.ssl.SSLHandshakeException;
+import javax.security.auth.x500.X500Principal;
+
+import static org.hamcrest.CoreMatchers.instanceOf;
+import static org.hamcrest.CoreMatchers.is;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+
+/**
+ * The test of SSL Netty layer with MTLS enabled.
+ */
+public class MtlsTest {
+
+    private static final Logger LOGGER = Logger.getLogger(MtlsTest.class.getName());
+
+    private static WebServer webServer;
+    private static WebClient clientWithoutCertificate;
+    private static WebClient clientWithCertificate;
+
+    /**
+     * Start the secured Web Server
+     *
+     * @param port the port on which to start the secured server; if less than 1,
+     *             the port is dynamically selected
+     * @throws Exception in case of an error
+     */
+    private static void startServer(int port) throws Exception {
+        HashMap<String, String> rawConfig = new HashMap<>();
+        rawConfig.put("client-auth", "REQUIRE");
+        rawConfig.put("trust.pem.certificates.resource.resource-path", "ssl/certificate.pem");
+        rawConfig.put("private-key.pem.key.resource.resource-path", "ssl/key.pkcs8.pem");
+        rawConfig.put("private-key.pem.cert-chain.resource.resource-path", "ssl/certificate.pem");
+
+        webServer = WebServer.builder(
+                Routing.builder()
+                        .any((req, res) -> {
+                            // This is annoyingly complex to pull just the CN out of an x509 cert, but it's generally easier if the caller
+                            // has access to other libraries like bouncy castle.
+                            Optional<X509Certificate> cert = req.context().get(WebServerTls.CLIENT_X509_CERTIFICATE, X509Certificate.class);
+                            res.send(cert.map(X509Certificate::getSubjectX500Principal).map(X500Principal::getName)
+                                .map(name -> Pattern.compile("(?:^|,\s?)(?:CN=(?<val>\"(?:[^\"]|\"\")+\"|[^,]+))").matcher(name))
```

Comment:
The pattern seems to be wrong - if you want to have \s as whitespace, it must be double escaped in Java (e.g. `\\s`). Probably true for other escapes in this regexp (and same in `master` in the original PR)

Reply:
Looking into this now. Have been a bit busy today. It's interesting that this causes a compilation failure on the 2.x branch (java 11?) but not on the master branch (java 17?). That's why I didn't catch it there 🤔

Reply:
This is fixed now.